zebra/zebra_routemap.c

```diff
@@ -359,26 +359,46 @@ ALIAS (no_match_ip_address_prefix_list,
 
 DEFUN (set_src,
        set_src_cmd,
-       "set src A.B.C.D",
+       "set src (A.B.C.D|X:X::X:X)",
        SET_STR
        "src address for route\n"
        "src address\n")
 {
   struct in_addr src;
+  struct in6_addr src6;
   struct interface *pif;
+  int is_ipv4;
```

Comment:
There're no bool type in the whole quagga codebase.
So I decided use their conventions.